src/IntelVectorMath.jl

```diff
@@ -8,6 +8,12 @@ const IVM = IntelVectorMath
 # import Base: .^, ./
 include("setup.jl")
 
+function __init__()
+    if Sys.isapple() && :SpecialFunctions in names(Main; imported = true)
+        @warn "It appears SpecialFunctions was loaded prior to this package, which currently on mac may lead to wrong results. For further details see github.com/JuliaMath/IntelVectorMath.jl"
```

Comment:
This will not catch the case when someone gets SpecialFunctions loaded as a dependency, because then the name will not be available in `Main`.

To do it "properly", you need to look at `Base.loaded_modules`

So it would be something like

```
special_functions_uuid = Base.UUID("276daf66-3868-5448-9aa4-cd146d93841b")
if Sys.isapple() && haskey(Base.loaded_modules, Base.PkgId(special_functions_uuid, "SpecialFunctions"))
    ...
end
```

And in fact, the problem (as shown in https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/700) is not really SpecialFunctions but CompilerSupportLibraries_jll so maybe that is the package that should be looked if it is loaded.

Reply:
Would you in general say this is reasonable thing to do?